src/stan/lang/ast/node/algebra_solver_control.hpp

```diff
@@ -0,0 +1,87 @@
+#ifndef STAN_LANG_AST_NODE_ALGEBRA_SOLVER_CONTROL_HPP
+#define STAN_LANG_AST_NODE_ALGEBRA_SOLVER_CONTROL_HPP
+
+#include <stan/lang/ast/node/expression.hpp>
+#include <string>
+
+namespace stan {
+  namespace lang {
+
+    struct expression;
+
+    /**
+     * Structure for an algebra solver statement with control
+     * parameters for the solver.
+     */
+    struct algebra_solver_control {
```

Comment:
This could've probably been implemented as a subclass of `algebra_solver` to reduce all the duplicated code.  We probably didn't do that for ODEs, either.